mongo/errors.go

```diff
@@ -286,6 +287,30 @@ func (we WriteError) Error() string {
 	return msg
 }
 
+// HasErrorCode returns true if the error has the specified code.
+func (we WriteError) HasErrorCode(code int) bool {
+	return int(we.Code) == code
+}
+
+// HasErrorLabel returns true if the error contains the specified label. WriteErrors do not contain labels,
+// so we inspect the message for the provided label instead.
```

Comment:
As the comment says, `WriteError` has no `Labels` field (unlike `CommandError`, `WriteException` and `BulkWriteException`), so `HasErrorLabel` might be a confusing API addition. There is, however, no way to implement `ServerError` without this function. Let me know what you think.,